### PR TITLE
Break apart "repo" pillar into "branch" and "repo"

### PR DESCRIPTION
**Pull request by rebecca-caktus**
There's a nifty command line trick for salt to pass in pillar data. If we break apart this pillar then we can easily override the branch in the `deploy` Fabric command.

See also:
https://github.com/caktus/django-project-template/pull/251
https://github.com/caktus/developer-documentation/pull/44

### CODE REVIEW

**Comment by mlavin** [comment]
This is going to be a bit painful to update but definitely for the best. @vkurup @dpoirier do you have any strong objection? This would enable us to get back to easy branch deployments as @rebecca-caktus has already proven on Hosted Tracpro.

File: project/repo.sls [hunk 1]
```diff
@@ -34,8 +34,8 @@ project_repo:
        - cmd: project_repo
   {% else %}
   git.latest:
-    - name: "{{ pillar['repo']['url'] }}"
-    - rev: "{{ pillar['repo'].get('branch', 'master') }}"
+    - name: "{{ pillar['repo'] }}"
+    - rev: "{{ pillar.get('branch', 'master') }}"
```
Comment (dpoirier) [inline project/repo.sls]: Could we maintain backward compatibility by using something like:

      - rev: "{{ pillar.get('branch', pillar['repo'].get('branch', 'master')) }}"

**Comment by dpoirier** [comment]
Maybe mention this in the developer docs, with a link to https://docs.saltstack.com/en/latest/topics/tutorials/pillar.html#setting-pillar-data-on-the-command-line and an example of using it to override the branch name on the fab command line?

**Comment by dpoirier** [comment]
I made a couple of comments but like the idea overall.

**Comment by rebecca-caktus** [comment]
Note: an alternative to this change would be to change the pillar merging strategy: https://docs.saltstack.com/en/latest/ref/configuration/master.html#pillar-source-merging-strategy

**Comment by rebecca-caktus** [comment]
See changes here: https://github.com/caktus/django-project-template/pull/251

**Comment by rebecca-caktus** [comment]
@dpoirier This is ready for re-review.

File: CHANGES.rst [hunk 2]
```diff
@@ -2,6 +2,16 @@ Margarita
...
 
+v 1.7.0 (unreleased)
+--------------------
+
+* Allow specifying the branch to deploy in the 'branch' pillar variable.
+    - Most projects should now be able to specify repo + url just once,
+      in `project.sls`.
+    - With this structure, the branch to deploy can now easily be overridden
+      from the command line. See Salt docs:
+      https://docs.saltstack.com/en/latest/topics/pillar/#set-pillar-data-at-the-command-line
```
Comment (dpoirier) [inline CHANGES.rst]: Maybe add an example command line here?  (Or if we have an example in the developer docs, link there?)

**Comment by dpoirier** [comment]
Let's update http://caktus.github.io/developer-documentation/margarita/states.html#project-repo along with this.

**Comment by rebecca-caktus** [comment]
@dpoirier Updated. 

Related pulls:
https://github.com/caktus/developer-documentation/pull/44
https://github.com/caktus/django-project-template/pull/251

**Comment by dpoirier** [comment]
:+1:

**Pull request merged**